packages/next-intl/src/extractor/formatters/utils.test.tsx

```diff
@@ -0,0 +1,28 @@
+import {describe, expect, it} from 'vitest';
+import type {ExtractedMessage} from '../types.js';
+import {getSortedMessages} from './utils.js';
+
+describe('getSortedMessages', () => {
+  it('uses message ids to break ties when reference paths match', () => {
+    const messages: Array<ExtractedMessage> = [
+      {id: 'beta', message: '', references: [{path: 'components/Footer.tsx'}]},
+      {id: 'alpha', message: '', references: [{path: 'components/Footer.tsx'}]},
+      {id: 'gamma', message: '', references: [{path: 'components/Header.tsx'}]}
+    ];
+
+    const sorted = getSortedMessages(messages).map((message) => message.id);
+
+    expect(sorted).toEqual(['alpha', 'beta', 'gamma']);
+  });
+
+  it('sorts by id when messages share the same inferred path', () => {
```

Comment:
@cursoragent Remove this test and instead add one that shows that we sort by reference and not be id if possible.